docs/build.sh

```diff
@@ -18,5 +18,24 @@
 # specific language governing permissions and limitations
 # under the License.
 
-rm -r _build
-make html
+set -e
+
+FWDIR="$(cd "`dirname "$0"`"; pwd)"
+cd "$FWDIR"
+
+[ -d _build ] && rm -r _build
+
+NUM_IGNORED_WARNINGS=4
+NUM_CURRENT_WARNINGS=$(make html |\
+    tee /dev/tty |\
+    grep 'build succeeded' |\
+    head -1 |\
+    sed -E 's/build succeeded, ([0-9]+) warnings\./\1/g')
+
+if [ "${NUM_CURRENT_WARNINGS}" != "${NUM_IGNORED_WARNINGS}" ]; then
+    echo
+    echo "Unexpected problems found in the documentation. "
```

Comment:
Could we make this log more descriptive? Currently there are no way to tell why / which part of the doc fails.

Reply:
Or at least print out how many current warning.

Reply:
the entire build log is displayed. `tee /dev/tty`. what can I show more?

Reply:
ok, I just scroll up and found the log. If you have time, do you want to fix the issue?

Reply:
It seems currently is 5 warning , compared to the 4 ignore warning.

Reply:
at the time of writing the PR it was 4.
if I find time, yes, but I do not have time yet. 
I have already corrected 450 errors in the documentation. these PR was to prevent the destruction of my previous work